feat: issue_06

yoshikawa-akr opened the pull request:
# [issue No.6](https://github.com/kenmori/handsonFrontend/blob/master/work/markup/1.md#6)

**issue**
5で作ったものを3秒後に解決されるようにしてください

## [CodeSandBox](https://codesandbox.io/s/amazing-forest-kcm9vg) / Latest commit: https://github.com/yoshikawa-akr/js-lessons/pull/6/commits/92044e8aa6aa6ae04895e4e4910a32161145c3eb

## Areas of concern in implementation
I am worried about naming variables and functions.

yoshikawa-akr committed:
feat: Added time difference feature for issue_06.

Implemented to be resolved in 3 seconds.

seeeen044 commented on issue_06/script.js:
```diff
@@ -11,21 +11,27 @@ const contents = [{
-    resolve(contents);
-});
-checkContents.then((value) => {
+const wait = (sec) => {
+    return new Promise((resolve) => {
+        setTimeout(resolve, sec * 1000);
+    });
+}
```
MUST
Resolve without receiving the contents.

yoshikawa-akr replied:
The process simply starts after 3 seconds, but the "contents" was not passed.
Fixed!
https://github.com/yoshikawa-akr/js-lessons/pull/6/commits/5e2cb0adcc2df2601b0b852975d3232d81cd26c5

seeeen044 commented on issue_06/script.js:
```diff
@@ -11,21 +11,27 @@ const contents = [{
+
     const fragment = document.createDocumentFragment();
-    for (let i = 0; i < value.length; i++) {
+    for (const item of contents) {
```
MUST
It refers to the variable contents instead of the resolved content.

yoshikawa-akr replied:
I solved the problem by setting the return value of PromiseSettings, which is being processed asynchronously, to a constant.
https://github.com/yoshikawa-akr/js-lessons/pull/6/commits/5e2cb0adcc2df2601b0b852975d3232d81cd26c5

---

Q. By the way, why do we refer to variables or constants rather than array names?

seeeen044 replied:
A.This is because if you write the array name, you will not receive the resolved value.

yoshikawa-akr replied:
I understood it to mean that I would receive a resolved value!
Thank you very much :sparkles:

seeeen044 commented on issue_06/script.js:
```diff
@@ -11,21 +11,27 @@ const contents = [{
+    });
+}
+
+async function createLists() {
+    await wait(3);
```
You can wait 3 seconds without writing this part.

```
  setTimeout(() => {
    resolve(contents);
  }, 3000);
})
```

yoshikawa-akr replied:
I couldn't get rid of the await description for the reason of waiting for the process in async.
However, I felt that setting extra arguments would make the code more difficult to read, so I changed the description as you suggested.
https://github.com/yoshikawa-akr/js-lessons/pull/6/commits/5e2cb0adcc2df2601b0b852975d3232d81cd26c5

seeeen044 replied:
nits

> createList.then((value) => {
  const fragment = document.createDocumentFragment();
  for (const item of value) {

If you write it like this, you will not need the async function.

csakyo replied:
@yoshikawa-akr 
Actually you should write this task without async await function. Because the async await function is in a much further issue. The challenge here is to get familiar with the Promise object. Could you try?

yoshikawa-akr replied:
Thanks for the comment!
I have corrected the points that you both pointed out.
https://github.com/yoshikawa-akr/js-lessons/pull/6/commits/92044e8aa6aa6ae04895e4e4910a32161145c3eb

csakyo commented on issue_06/script.js:
```diff
@@ -11,21 +11,27 @@ const contents = [{
-    resolve(contents);
-});
-checkContents.then((value) => {
+const wait = (sec) => {
```
IMO
I don't think this constant name is appropriate. This is because waiting is not the goal here.

yoshikawa-akr replied:
You're right.
I made some changes, but I'm not too sure. 
https://github.com/yoshikawa-akr/js-lessons/pull/6/commits/163ba4c8cef7a3219f8be4e7f5cc6b62c2764fda

If there is something more appropriate, please let me know.

csakyo replied:
It is OK for me! I think the goal here is to get the data in the Promise function. So, ``getData`` or ``getList`` etc.. may be fine. It's just an idea, no need to modify it！

yoshikawa-akr committed:
refactor: Adding arguments.

yoshikawa-akr committed:
refactor: Changing variable names.

Change from wait to promiseSetting.

yoshikawa-akr committed:
refactor: Delete the arguments.

Delete the arguments sec and value for promiseSettings and target for createLists. Set the return value of promiseSettings as a variable.

seeeen044 commented on issue_06/script.js:
```diff
@@ -11,21 +11,29 @@ const contents = [{
-    resolve(contents);
-});
-checkContents.then((value) => {
+const promiseSettings = () => {
+    return new Promise((resolve) => {
```
```suggestion
const promiseSettings = () => {
const contents = [{
    to: "bookmark.html",
    img: "1.png",
    alt: "画像1",
    text: "ブックマーク"
}, {
    to: "message.html",
    img: "2.png",
    alt: "画像2",
    text: "メッセージ"
}];
    return new Promise((resolve) => {
```
nits
This will make it easier to read.

yoshikawa-akr replied:
I found this one easier to read, so I fixed it.
https://github.com/yoshikawa-akr/js-lessons/pull/6/commits/92155ae3c404d109745778dc5f2f2c653f615390

yoshikawa-akr committed:
refactor: Changing the array description position.

yoshikawa-akr committed:
refactor: Remove async/await and replace with Promise.then.
